libcxx/CMakeLists.txt

```
@@ -424,6 +428,8 @@ set(LIBCXX_INSTALL_INCLUDE_DIR "${CMAKE_INSTALL_INCLUDEDIR}/c++/v1" CACHE STRING
     "Path where target-agnostic libc++ headers should be installed.")
 set(LIBCXX_INSTALL_RUNTIME_DIR "${CMAKE_INSTALL_BINDIR}" CACHE STRING
     "Path where built libc++ runtime libraries should be installed.")
+set(LIBCXX_INSTALL_MODULE_DIR "usr/share/libc++/v1" CACHE STRING
+    "Path where target-agnostic libc++ modules should be installed.")
```

I think the installation location needs a better default. For myself, I build with `cmake -DCMAKE_INSTALL_PREFIX=/my/own/path` and a `usr/share` scheme doesn't fit in with what that produces.
```suggestion
set(LIBCXX_INSTALL_MODULE_INTERFACE_UNIT_SOURCE_DIR "usr/share/libc++/v1" CACHE STRING
    "Path where libc++ C++20 module interface unit source files should be installed.")
```

---

From https://cmake.org/cmake/help/v3.28/module/GNUInstallDirs.html, I would suggest `${CMAKE_INSTALL_DATAROOTDIR}/libc++/v1`, which would be `$prefix/share/libc++/v1`. Vendors would generally set `CMAKE_INSTALL_PREFIX=/usr`, which would give `/usr/share/libc++/v1`.

---

I did not add the "C++20", that can be confusing.
C++20 added modules
C++23 added the `std` and `std.compat` module, (which the major vendors make available in C++20).